File: app/src/main/java/club/electro/ui/user/UserProfileViewModelFactory.java
package club.electro.ui.user;

import android.app.Application;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class UserProfileViewModelFactory implements ViewModelProvider.Factory {
    private Application mApplication;
    private Long mUserId;

    public UserProfileViewModelFactory(Application application, Long userId) {
        mApplication = application;
        mUserId = userId;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new UserProfileViewModel(mApplication, mUserId);
    }
}

File: app/src/main/java/club/electro/ui/user/UserProfileViewModelFactory.kt
package club.electro.ui.user

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class UserProfileViewModelFactory(
    private val mApplication: Application,
    private val mUserId: Long
) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return UserProfileViewModel(mApplication, mUserId) as T
    }
}